fletcher/algorithms/string.py

```diff
@@ -267,6 +268,72 @@ def _text_contains_case_sensitive(data: pa.Array, pat: str) -> pa.Array:
     )
 
 
+@njit
+def _get_zstring(width: int, buffer: np.ndarray) -> np.ndarray:
+    num_zeroes = width
+    for i in np.bitwise_xor(np.right_shift(buffer, 6), 2):
+        if i != 0:
+            num_zeroes -= 1
+    if num_zeroes <= 0:
+        return buffer
+    # Create array of byte-representation of 0s concat'd with buffer
+    return np.concatenate((np.full(num_zeroes, 48, dtype=np.uint8), buffer), axis=0)
```

Comment:
I would expect this line to be quite compute-heavy as you allocate space several times. A possibly more performant alternative would be to have a local buffer that is reused between each row. You only need to allocate that once as the upper bound of the size is `4 * width`. Any strings longer than that don't get zero-filled, smaller ones get zero-filled but won't be larger than `4 * width` after being zero-filled. 

With this trick, no more temporary allocations should be possible.

Reply:
We were just about to ask you for some additional suggestions so will go ahead and work on this. Much appreciated! 😄 @ArseniiDunaevQC